Chore/update electron builder

syns2191 opened the pull request:
# PR

- [x] Have you followed the [contributing guidelines](https://github.com/ever-co/ever-gauzy/blob/master/.github/CONTRIBUTING.md)?
- [x] Have you explained what your changes do, and why they add value?

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

---

dependabot[bot] committed:
chore(deps): bump electron from 30.5.1 to 35.7.5

Bumps [electron](https://github.com/electron/electron) from 30.5.1 to 35.7.5.
- [Release notes](https://github.com/electron/electron/releases)
- [Changelog](https://github.com/electron/electron/blob/main/docs/breaking-changes.md)
- [Commits](https://github.com/electron/electron/compare/v30.5.1...v35.7.5)

---
updated-dependencies:
- dependency-name: electron
  dependency-version: 35.7.5
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

syns2191 committed:
chore: update electron-builder and better-sqlite

syns2191 committed:
chore: added local dependencies patcher to resolve desktop local dependencies package

syns2191 committed:
fix: script build destkop

evereq commented:
@syns2191 see more AI suggestions please

syns2191 committed:
fix: clean typo and resolve ai suggestion

syns2191 committed:
fix: remove unused fallback empty string

evereq commented:
@syns2191 I did "changed the base branch from dependabot/npm_and_yarn/electron-35.7.5 to develop", hope it's OK? This PR should target develop branch and if that other PR that update only part of electron should be merged, please change THAT PR target to be this feature branch, i.e. for PR from bot, change target to be `chore/update-electron-builder`! Not other way around how it was...

Basically, can you make sure ALL required changes are in one PR and it's this PR please? 

Also, it feels this PR might make a LOT of issues with builds for all our apps, so it will be great to get @adkif involved to test everything on his Mac before we can merge this PR. But FIRST, let's make this PR to be fully ready and include all changes etc.

adkif committed:
build(desktop): update build scripts and remove unused prepare command

The `build:desktop-timer:mac:quick` script now includes `build:desktop-timer`
and `patch:desktop-timer:local:deps` to ensure all necessary prerequisites
are met before building the Electron package.
The `prepare:desktop:dev` script was removed as it is no longer used for
desktop development.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ever-co/ever-gauzy?pullRequest=9120) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/ever-co/ever-gauzy?pullRequest=9120) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: adkif<br/>:white_check_mark: syns2191<br/>:white_check_mark: evereq<br/>:x: dependabot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ever-co/ever-gauzy?pullRequest=9120) it.</sub>

syns2191 committed:
Merge branch 'develop' of github.com:ever-co/ever-gauzy into chore/update-electron-builder

syns2191 committed:
fix: cspell variable name

syns2191 commented:
> @syns2191 I did "changed the base branch from dependabot/npm_and_yarn/electron-35.7.5 to develop", hope it's OK? This PR should target develop branch and if that other PR that update only part of electron should be merged, please change THAT PR target to be this feature branch, i.e. for PR from bot, change target to be `chore/update-electron-builder`! Not other way around how it was...
> 
> Basically, can you make sure ALL required changes are in one PR and it's this PR please?
> 
> Also, it feels this PR might make a LOT of issues with builds for all our apps, so it will be great to get @adkif involved to test everything on his Mac before we can merge this PR. But FIRST, let's make this PR to be fully ready and include all changes etc.

@evereq it's ok you change the target branch to develop. you can close the other branch / dependabot pr. all the upgrade electron and electron-builder is in this PR.

evereq commented on packages/desktop-lib/package.json:
```diff
@@ -34,7 +34,7 @@
 		"@gauzy/desktop-activity": "^0.1.0",
 		"get-windows": "^9.2.3",
-		"better-sqlite3": "9.6.0",
+		"better-sqlite3": "12.2.0",
```
@syns2191 @adkif  I 100% remember there were some issues with `better-sqlite3` if we update it, that's why we "pin" version.... We need to carefully test builds for all OSs and how it runs with this update

evereq commented:
@syns2191 as part of this update, did you updated packages / code etc and TESTED it all in ALL Destkop Apps we have? E.g. what about Desktop API Server, Desktop MCP and so on? We need to make sure if we update electron / electron-builder, we have SAME versions in ALL apps! Also, what OS you test it all exactly?

syns2191 committed:
Merge branch 'develop' of github.com:ever-co/ever-gauzy into chore/update-electron-builder

syns2191 committed:
chore: update electron from 35.7.5 to 38.1.2 and electron builder from 25.1.8 to 26.0.3

syns2191 committed:
chore: add mcp server desktop patch:local:deps before run electron-builder

syns2191 committed:
chore: add mcp server desktop patch:local:deps before run electron-builder

syns2191 committed:
fix: ai suggestion

syns2191 committed:
fix: ai suggestion

syns2191 committed:
chore: update yarn.lock

evereq committed:
Merge pull request #9150 from ever-co/develop

latest

evereq commented:
@syns2191 please rebase on top of latest on develop and fix merge conflicts etc

syns2191 committed:
Merge branch 'develop' of github.com:ever-co/ever-gauzy into chore/update-electron-builder

syns2191 committed:
chore: update yarn.lock

syns2191 committed:
chore: update cspell

evereq committed:
Merge pull request #9168 from ever-co/develop

latest

syns2191 committed:
Merge branch 'develop' of github.com:ever-co/ever-gauzy into chore/update-electron-builder

syns2191 committed:
Merge branch 'develop' of github.com:ever-co/ever-gauzy into chore/update-electron-builder

syns2191 committed:
chore: update yarn.lock

syns2191 committed:
Merge branch 'develop' of github.com:ever-co/ever-gauzy into chore/update-electron-builder

syns2191 committed:
fix: desktop integrated server failed to started

syns2191 committed:
fix: desktop integrated server failed to started

syns2191 committed:
fix: desktop build using new electron-builder

syns2191 committed:
Merge branch 'fix/desktop-integrated-server-not-started' of github.com:ever-co/ever-gauzy into fix/build-all-desktop

syns2191 committed:
fix: remove unnecessary custom build script

syns2191 committed:
fix: revert back unrelated change

syns2191 committed:
Merge branch 'develop' of github.com:ever-co/ever-gauzy into chore/update-electron-builder
